index.js

```diff
@@ -143,16 +144,12 @@ export function execa(rawFile, rawArgs, rawOptions) {
 }
 
 const handlePromise = async ({spawned, options, context, stdioStreams, command, escapedCommand}) => {
-	const spawnedPromise = getSpawnedPromise(spawned);
-	const timedPromise = setupTimeout(spawned, options, spawnedPromise);
-	const processDone = setExitHandler(spawned, options, timedPromise);
```

Comment:
Instead of manipulating promises like this, put all this logic in the same function using `Promise.race()`/`Promise.all()` (see below).